### DISCUSSION TITLE
EnOcean PTM 215Z hold button

### DISCUSSION THREAD

**Discussion by Blunkfield** [Koenkk/zigbee2mqtt#18592] [Devices]
New feature

Hello,

I'm using the EnOcean PTM 215Z module and I see that it's only possible to retrieve the "press" of a button. I would need to retrieve the hold to be able to dim lamps. Is it possible to add this function?

Best regards.

Nicolas Vauthey

**Comment by ooglyhLL** [comment]
You will have to solve this at a higher level, e.g. in Node-RED with a “button-events” node (https://github.com/bartbutenaers/node-red-contrib-button-events).

**Comment by chris-1243** [comment]
If you use Home assistant, I would suggest to use a blueprint which works perfectly well for me [https://gist.github.com/vandalon/d2a327297d579b304549e96c66765992](url).

You need to enable the elapsed function in zigbee2mqtt.

Otherwise, you need to find a way to count the time between a press and release command sent by the controller (the elapsed function is used that way...).

Or as suggested via Node-RED
